packages/lesswrong/server/languageModels/postSummaryResolver.ts

```diff
@@ -0,0 +1,27 @@
+import { Posts } from '../../lib/collections/posts/collection';
+import { augmentFieldsDict } from '../../lib/utils/schemaUtils'
+import { dataToMarkdown } from '../editor/conversionUtils';
+import { languageModelGenerateText } from './languageModelIntegration';
+
+augmentFieldsDict(Posts, {
+  languageModelSummary: {
+    resolveAs: {
+      type: "String",
+      resolver: async (post: DbPost, args: void, context: ResolverContext): Promise<string> => {
+        const markdownPostBody = dataToMarkdown(post.contents?.originalContents?.data, post.contents?.originalContents?.type);
+        const authorName = "Authorname"; //TODO
+        
+        return await languageModelGenerateText({
```

Comment:
nit: `return await` is redundant

Reply:
I actually think including the `await` is better. It makes the fact that it's something that can wait explicit.

Reply:
If the issue is the lack of language server integration, I'd rather explicitly annotate the function's return type to say `Promise<...>`, but 🤷

Reply:
It's not a language server integration thing. It seems better to me to just consistently put `await` in front of calls to async functions since it makes it possible to look at a function and spot all the places that might be waiting on the network, and it's a simpler convention to follow.